plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/templates/CodetemplatesGeneratorFragment2.xtend

```diff
@@ -0,0 +1,123 @@
+/*******************************************************************************
+ * Copyright (c) 2015 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.xtext.generator.ui.templates
+
+import com.google.inject.Inject
+import java.util.Collection
+import java.util.Collections
+import org.eclipse.xtend2.lib.StringConcatenationClient
+import org.eclipse.xtext.AbstractRule
+import org.eclipse.xtext.Grammar
+import org.eclipse.xtext.util.PolymorphicDispatcher
+import org.eclipse.xtext.xtext.generator.AbstractGeneratorFragment2
+import org.eclipse.xtext.xtext.generator.XtextGeneratorNaming
+import org.eclipse.xtext.xtext.generator.model.FileAccessFactory
+import org.eclipse.xtext.xtext.generator.model.GuiceModuleAccess
+import org.eclipse.xtext.xtext.generator.model.TypeReference
+import org.eclipse.xtext.xtext.generator.parser.antlr.GrammarNaming
+
+import static extension org.eclipse.xtext.GrammarUtil.*
+import static extension org.eclipse.xtext.xtext.generator.model.TypeReference.typeRef
+
+/**
+ * Contributes the 'Partial...ContentAssisParser' and performs further
+ *  Guice-based registrations. Does not contribute 'plugin.xml' entries.
+ *  
+ *
+ * @author Christian Schneider - Initial contribution and API
+ */
+class CodetemplatesGeneratorFragment2 extends AbstractGeneratorFragment2 {
+
+	@Inject
+	extension GrammarNaming
+
+	@Inject
+	extension XtextGeneratorNaming
+
+	@Inject
+	FileAccessFactory fileAccessFactory
+
+	def TypeReference getPartialContentAssistParserClass(Grammar g) {
+		new TypeReference(g.genericIdeBasePackage + ".contentassist.antlr", "Partial" + g.simpleName + "ContentAssistParser")
+	}
+
+	override generate() {
+		if (projectConfig.eclipsePlugin?.manifest != null) {
+			projectConfig.eclipsePlugin?.manifest.requiredBundles += #[
+				"org.eclipse.xtext.ui", "org.eclipse.xtext.ui.codetemplates.ui"
+			]
+		}
+		
+		new GuiceModuleAccess.BindingFactory()
+				.addTypeToProviderInstance(
+					"org.eclipse.xtext.ui.codetemplates.ui.preferences.TemplatesLanguageConfiguration".typeRef,
+					'''org.eclipse.xtext.ui.codetemplates.ui.AccessibleCodetemplatesActivator.getTemplatesLanguageConfigurationProvider()''')
```

Comment:
ah ... maybe I got it now. You mean instead of hardcoding
```
org.eclipse.xtext.ui.codetemplates.ui.AccessibleCodetemplatesActivator...
```
I better use ``«new TypeRef(org.eclipse.xtext.ui.codetemplates.ui.AccessibleCodetemplatesActivator)»....``
That's true.